c7n/filters/kms.py

```diff
@@ -67,7 +67,10 @@ def get_related(self, resources):
     def get_related_ids(self, resources):
         related_ids = super().get_related_ids(resources)
         normalized_ids = []
+        alias_to_id = self.key_alias_to_key_id()
```

Comment:
I made this change so we can turn the `kms-key` resource's `alias_map` inside out once before looping through resources.

Reply:
Nice, yup this seems to be better since it can also be reused when needed